### PR TITLE
Add Hibernate 5 build

### LINKED ISSUE #1621: HibernateDomainExporter not compatible with Hibernate 5 Configuration
Using Hibernate 5.0.2 and QueryDsl Tool 4.0.5, we have the following Problem.  The Hibernate 5 Configuration object no longer has the getMappedSuperclassMappings method.

```
at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:183)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
    ... 26 more
Caused by: org.apache.maven.plugin.PluginContainerException: An API incompatibility was encountered while executing: ****

SNAPSHOT:generate: java.lang.NoSuchMethodError: org.hibernate.cfg.Configuration.getMappedSuperclassMappings()Ljava/util/Iterator;
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:181)
    ... 27 more
Caused by: java.lang.NoSuchMethodError: org.hibernate.cfg.Configuration.getMappedSuperclassMappings()Ljava/util/Iterator;
    at com.querydsl.jpa.codegen.HibernateDomainExporter.collectTypes(HibernateDomainExporter.java:148)
    at com.querydsl.jpa.codegen.AbstractDomainExporter.execute(AbstractDomainExporter.java:116)
    at ch.bedag.gba.maven.plugin.querydsl.HibernateExportMojo.execute(HibernateExportMojo.java:62)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
    ... 27 more
```

### PR DESCRIPTION
**Pull request by timowest**

### CODE REVIEW

**Comment by johnktims** [comment]
Maybe we should create multiple modules to support the major versions of hibernate. Hibernate keeps moving the interfaces and classes around, and supporting all versions in one module allows users to create queries that might not be supported in the version of hibernate they're using.

**Comment by timowest** [comment]
Yes. That might be necessary in this case.

File: querydsl-jpa/src/test/java/com/querydsl/jpa/support/TeradataDialect.java [hunk 1]
```diff
@@ -304,41 +297,42 @@ public boolean supportsBindAsCallableArgument() {
...
+//
+//        /**
+//         * Extract the name of the violated constraint from the given
+//         * SQLException.
+//         *
+//         * @param sqle
+//         *            The exception that was the result of the constraint
+//         *            violation.
+//         * @return The extracted constraint name.
+//         */
```
Comment (Shredder121) [inline querydsl-jpa/src/test/java/com/querydsl/jpa/support/TeradataDialect.java]: Commented code in the repository?

File: .travis.yml [hunk 2]
```diff
@@ -7,17 +7,18 @@ addons:
   postgresql: "9.3"
 env:
   matrix:
-    - MODULE=!querydsl-scala
-    - MODULE=querydsl-scala
+    - MODULE=!querydsl-scala PROFILES=all,travis,examples
+    - MODULE=!querydsl-jpa-codegen PROFILES=jpa,hibernate5,travis,examples
```
Comment (Shredder121) [inline .travis.yml]: Shouldn't this be called `hibernate5`?
Or what do you have in mind?

Comment (Shredder121) [inline .travis.yml]: Or hmm, I forgot that it is a `--projects` pattern.
It seems to work nevermind.

Comment (timowest) [inline .travis.yml]: No, this was intended.

Comment (Shredder121) [inline .travis.yml]: I thought it would run more than necessary, but it doesn't have `all` so it's fine.
Carry on :wink:

**Comment by Shredder121** [comment]
This is good.
So this is just the build?

**Comment by timowest** [comment]
Yes, this is only the build part.

File: pom.xml [hunk 3]
```diff
@@ -52,6 +52,11 @@
     <teradata.version>13.10.00.35</teradata.version>
     <firebird.version>2.2.5</firebird.version>
 
+    <!-- JPA deps -->
+    <hibernate.version>4.3.11.Final</hibernate.version>
+    <hibernate.validator.version>4.3.1.Final</hibernate.validator.version>
+    <eclipselink.version>2.6.0-RC1</eclipselink.version>
```
Comment (johnktims) [inline pom.xml]: 2.6.2 is available now, so we don't have to use the RC anymore.

**Comment by timowest** [comment]
@johnktims Is this now ok?

**Comment by johnktims** [comment]
Yes.

**Pull request merged**